### PR TITLE
[Feature][Connector-V2] connector-kafka source support data conversion extracted by kafka connect source

### PR DESCRIPTION
**Pull request by sunxiaojian**
## Purpose of this pull request

#4137

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by sunxiaojian** [comment]
@hailin0 @TyrantLucifer

**Comment by TyrantLucifer** [comment]
> @hailin0 @TyrantLucifer

Hi @sunxiaojian , Overall LGTM, thank you for your contribution of Apache SeaTunnel, but unfortunately this current pull request overlaps in functionality with #3981, don't worry, your proposed solution and implementation approach are better. This pull request can be merged after #3981. Please be patient and wait for good news. Once there are new updates, I will ping you on this page.

**Comment by EricJoy2048** [comment]
@hailin0  PTAL

**Comment by hailin0** [comment]
@sunxiaojian please resolve the conflict

**Comment by sunxiaojian** [comment]
> @sunxiaojian please resolve the conflict



> @sunxiaojian please resolve the conflict

@hailin0 done

**Comment by sunxiaojian** [comment]
> @sunxiaojian please resolve the conflict

@hailin0 done

**Comment by hailin0** [comment]
> > @sunxiaojian please resolve the conflict
> 
> @hailin0 done

@sunxiaojian 
Duplicate with https://github.com/apache/seatunnel/pull/5066 feature?

File: seatunnel-connectors-v2/connector-kafka/pom.xml [hunk 1]
```diff
@@ -61,6 +62,33 @@
...
+                <exclusion>
+                    <groupId>org.glassfish.jersey.core</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-connector-mongodb</artifactId>
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-kafka/pom.xml]: Depend on mongodb connect?

File: seatunnel-connectors-v2/connector-kafka/pom.xml [hunk 2]
```diff
@@ -61,6 +62,33 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.apache.seatunnel</groupId>
+            <artifactId>seatunnel-format-compatible-connect-json</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-embedded</artifactId>
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-kafka/pom.xml]: Depend on debezium-embedded?

Should it be moved to `seatunnel-format-compatible-connect-json`

**Comment by sunxiaojian** [comment]
> > > @sunxiaojian please resolve the conflict
> > 
> > 
> > @hailin0 done
> 
> @sunxiaojian Duplicate with #5066 feature?

It is also compatible with Debezium and can be removed

**Comment by sunxiaojian** [comment]
> > > @sunxiaojian please resolve the conflict
> > 
> > 
> > @hailin0 done
> 
> @sunxiaojian Duplicate with #5066 feature?

@hailin0  It is also compatible with debezium and can be removed

**Comment by hailin0** [comment]
check ci

File: seatunnel-formats/seatunnel-format-compatible-connect-json/pom.xml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-embedded</artifactId>
+            <version>${debezium.version}</version>
+            <exclusions>
+                <exclusion>
```
Comment (hailin0) [inline seatunnel-formats/seatunnel-format-compatible-connect-json/pom.xml]: check all Dependencies
https://mvnrepository.com/artifact/io.debezium/debezium-embedded/1.6.4.Final

File: seatunnel-formats/seatunnel-format-compatible-connect-json/pom.xml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                    <groupId>org.glassfish.jersey.core</groupId>
+                    <artifactId>*</artifactId>
+                </exclusion>
+            </exclusions>
+        </dependency>
+
+        <dependency>
+            <groupId>io.debezium</groupId>
+            <artifactId>debezium-connector-mongodb</artifactId>
+            <version>${debezium.version}</version>
```
Comment (hailin0) [inline seatunnel-formats/seatunnel-format-compatible-connect-json/pom.xml]: exclud all dependencies?

https://mvnrepository.com/artifact/io.debezium/debezium-connector-mongodb/1.6.4.Final
image

**Comment by sunxiaojian** [comment]
@hailin0 PTAL

**Comment by sunxiaojian** [comment]
CI error is not related to this modification

**Review by hailin0 — APPROVED** [review]

**Review by TyrantLucifer — APPROVED** [review]

**Pull request merged**